### ISSUE TITLE
SSL handshake error when connecting to cluster ingress when telepresence is connected

### ISSUE THREAD

**Issue by Gikkman** [telepresenceio/telepresence#2274]
**Describe the bug**
When I connect to the cluster using Telepresence, I can no longer send requests to external ingresses of that cluster, due to failed SSL handshakes. We're using Openshift 4, with ingresses that has SSL configured using cluster-wide certificates (using Openshift annotations).

**To Reproduce**
 At first, this works just fine:
```
$ curl -v https://test-service-ingress.namespace.my-cluster.net/

*   Trying 10.17.227.11...
* TCP_NODELAY set
* Connected to test-service-ingress.namespace.my-cluster.net (10.17.227.11) port 443 (#0)
* ALPN, offering http/1.1
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* successfully set certificate verify locations:
*   CAfile: /usr/local/php5/ssl/cert.pem
  CApath: none
* TLSv1.2 (OUT), TLS header, Certificate Status (22):
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server did not agree to a protocol
* Server certificate:
*  subject: CN=*.eph.my-cluster.net
*  start date: Sep 10 00:00:00 2021 GMT
*  expire date: Sep 10 23:59:59 2022 GMT
*  subjectAltName: host "test-service-ingress.namespace.my-cluster.net" matched cert's "*.eph.my-cluster.net"
*  issuer: C=GB; ST=Greater Manchester; L=Salford; O=Sectigo Limited; CN=Sectigo RSA Domain Validation Secure Server CA
*  SSL certificate verify ok.
> GET / HTTP/1.1
> Host: test-service-ingress.namespace.my-cluster.net
> User-Agent: curl/7.54.0
> Accept: */*
> 
< HTTP/1.1 200 OK
< server: nginx/1.21.0
< date: Wed, 15 Dec 2021 14:13:06 GMT
```

Then I connect with telepresence:
```
$ telepresence connect

Launching Telepresence Root Daemon
Need root privileges to run: /usr/local/bin/telepresence daemon-foreground /Users/gikkman/Library/Logs/telepresence '/Users/gikkman/Library/Application Support/telepresence' ''
Password:
Launching Telepresence User Daemon
Connected to context eph-test/api-context-platform-net:6443/gikkman@companygroup.com (https://api.my-cluster.net:6443)
```

And then we try to read the same ingress address again:
```
$ curl -v https://test-service-ingress.namespace.my-cluster.net/

*   Trying 10.17.227.11...
* TCP_NODELAY set
* Connected to test-service-ingress.namespace.my-cluster.net (10.17.227.11) port 443 (#0)
* ALPN, offering http/1.1
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* successfully set certificate verify locations:
*   CAfile: /usr/local/php5/ssl/cert.pem
  CApath: none
* TLSv1.2 (OUT), TLS header, Certificate Status (22):
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to test-service-ingress.namespace.my-cluster.net:443 
* Closing connection 0
curl: (35) OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to test-service-ingress.namespace.my-cluster.net:443 
```

**Expected behavior**
I'd expect SSL handshakes to work, irregardless of whether I got telepresence running or not. It might be that my understanding of how telepresence works is insufficient, but I thought requests that were targeted at the exluded TLDs (.com, .net, .org, .io and so on) didn't go via the telepresence proxy. Hence why I am a bit surprised at this being an issue. 

**Versions (please complete the following information):**
```
$ telepresence version

Client: v2.4.9 (api v3)
Root Daemon: v2.4.9 (api v3)
User Daemon: v2.4.9 (api v3)
```

**Additional context**
Requests to the rest of the internet works just fine (e.g., https://www.google.com) and even to the cluster's own APIs. I can also connect to serivces using their service addresses just fine while connected with telepresence (e.g., http://my-service.namespace.svc).

We've confirmed this behaviour on both my laptor (MacOS 10.15.7) and a colleague (Windows 10), so it's not isolated to my computer. It could of course be something in our cluster's config, but I am unsure what that'd be.

**Comment by thallgren** [comment]
@Gikkman a couple of questions:
1. The service that you are curling, is it using a workload that has a Telepresence traffic-agent installed? If that's the case, does a `telepresence uninstall --agent <workload name>` make a difference?
2. If that's not the case, does this affect all services in the cluster that uses TLS?
3. Are you logged in to telepresence when this happens?
4. Would it be possible for you to provide the result of doing:
    1. `telepresence loglevel debug` 
    2. curl your service (and fail)
    3. `telepresence gather logs`

**Comment by Gikkman** [comment]
1. The service doesn't have any traffic-agent installed.
2. I am not 100% sure what you mean here. I can still reach the cluster in general, say via `kubectl get projects` or by curling the cluster's API endpoint, even if they use TLS. It seems to be just the ingresses.
3. No. As far as I understand it, we're not using Ambassador Cloud on our cluster.
4. Sure, attaching the logs:

[telepresence_logs.zip](https://github.com/telepresenceio/telepresence/files/7854989/telepresence_logs.zip)

**Comment by Gikkman** [comment]
This seems to have been solved when moving to 2.6.X so I'll close this issue
